ark/utils/test_utils.py

```diff
@@ -0,0 +1,143 @@
+import os
+import numpy as np
+import xarray as xr
+import skimage.io as io
+
+from mibidata import mibi_image as mi, tiff
+
+# required metadata for mibitiff writing (barf)
+MIBITIFF_METADATA = {
+    'run': '20180703_1234_test', 'date': '2017-09-16T15:26:00',
+    'coordinates': (12345, -67890), 'size': 500., 'slide': '857',
+    'fov_id': 'Point1', 'fov_name': 'R1C3_Tonsil',
+    'folder': 'Point1/RowNumber0/Depth_Profile0',
+    'dwell': 4, 'scans': '0,5', 'aperture': 'B',
+    'instrument': 'MIBIscope1', 'tissue': 'Tonsil',
+    'panel': '20170916_1x', 'mass_offset': 0.1, 'mass_gain': 0.2,
+    'time_resolution': 0.5, 'miscalibrated': False, 'check_reg': False,
+    'filename': '20180703_1234_test', 'description': 'test image',
+    'version': 'alpha',
+}
+
+
+def _gen_tif_data(fov_number, chan_number, img_shape, fills, dtype):
+    if not fills:
+        tif_data = np.random.randint(0, 100,
+                                     size=(fov_number, *img_shape, chan_number)).astype(dtype)
+    else:
+        tif_data = np.full(
+            (*img_shape, fov_number, chan_number),
+            (np.arange(fov_number * chan_number) % 256).reshape(fov_number, chan_number),
+            dtype=dtype
+        )
+        tif_data = np.moveaxis(tif_data, 2, 0)
+
+    return tif_data
+
+
+def _create_tifs(base_dir, fov_names, img_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(img_names), shape, fills, dtype)
+
+    if sub_dir is None:
+        sub_dir = ""
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        filelocs[fov] = []
+        fov_path = os.path.join(base_dir, fov, sub_dir)
+        os.makedirs(fov_path)
+        for j, name in enumerate(img_names):
+            io.imsave(os.path.join(base_dir, name), tif_data[i, :, :, j])
+            filelocs[fov].append(os.path.join(base_dir, fov, sub_dir, name))
+
+    return filelocs, tif_data
+
+
+def _create_multitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
```

Comment:
should we check for sub_dir here similar to _create_tif one?

Reply:
in `_create_tif`, we are loading single channel tifs, where each tifs represents a channel, and a whole folder represents an fov.  The channel images are typically stored in subdirectories (often called 'IMGS', 'imgs', etc.) so the subdir parameter allows us to find the channel data.

In multi/mibitiffs, each tiff represents a fov, while individual layers within a tiff represent the channels, so there's no need to have a subdir argument.

Also, I'm pretty sure the multitiffs are only being generated as the inputs into deepcell (only mibitiff and single tifs are taken in by preprocessings) so we have almost full control for the 'standard' way these are organized in a file structure.